app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/splash/SplashScreenTest.kt

```diff
@@ -119,34 +119,41 @@ class SplashScreenTest : BaseAndroidComposeTest() {
     }
 
     @Test
-    fun when_with_incorrect_email_and_password_error_dialog_is_shown() {
+    fun when_with_correct_email_and_password_goes_to_home() {
         with(composeTestRule) {
+            waitForIdle()
             onNodeWithContentDescription(getString(R.string.login_email_text_field))
                 .performTextInput(EMAIL)
             onNodeWithContentDescription(getString(R.string.login_password_text_field))
-                .performTextInput(INVALID_PASSWORD)
+                .performTextInput(VALID_PASSWORD)
             onNodeWithContentDescription(getString(R.string.login_log_in_button))
                 .performClick()
-            onNodeWithText(ERROR_MESSAGE).assertIsDisplayed()
+            waitForIdle()
+            assertTrue(splashViewModel.isLoginSuccess.value)
         }
     }
 
     @Test
-    fun when_with_correct_email_and_password_goes_to_home() {
+    fun when_with_incorrect_email_and_password_error_dialog_is_shown() {
         with(composeTestRule) {
+            waitForIdle()
             onNodeWithContentDescription(getString(R.string.login_email_text_field))
                 .performTextInput(EMAIL)
             onNodeWithContentDescription(getString(R.string.login_password_text_field))
-                .performTextInput(VALID_PASSWORD)
+                .performTextInput(INVALID_PASSWORD)
             onNodeWithContentDescription(getString(R.string.login_log_in_button))
                 .performClick()
-            waitForIdle()
-            assert(splashViewModel.isLoginSuccess.value)
+            onNodeWithText(ERROR_MESSAGE).assertIsDisplayed()
         }
     }
 
     private fun setupSplashComposeRule() {
-        every { preferenceManager.getStringData(PREF_REFRESH_TOKEN) } returns ""
+        every { preferenceManager.getStringData(any()) } returns ""
+        splashViewModel = SplashViewModel(
+            loginRepo = FakeLoginRepo(),
```

Comment:
Couldn't we inject `FakeLoginRepo` too similar to `PreferenceManager`? 🤔

Reply:
We could. I just want to make a sample here that if we are going to use test doubles, we could take a look at this. 😬